src/overlays/actors/ovl_En_Zl1/z_en_zl1.c

```diff
@@ -15,51 +15,650 @@ void EnZl1_Destroy(Actor* thisx, GlobalContext* globalCtx);
+        this->unk_1E6 = 0;
+        this->actionFunc = func_80B4BC78;
+    } else if ((Flags_GetEventChkInf(9)) && (Flags_GetEventChkInf(0x25)) && (Flags_GetEventChkInf(0x37))) {
+        Actor_Kill(thisx);
+    } else if (((Flags_GetEventChkInf(9)) && (Flags_GetEventChkInf(0x25))) ||
+               ((Flags_GetEventChkInf(9)) && (Flags_GetEventChkInf(0x37)))) {
+        frameCount = SkelAnime_GetFrameCount(&D_06000438.genericHeader);
+        SkelAnime_ChangeAnim(&this->skelAnime, &D_06000438, 1.0f, 0.0f, frameCount, 0, 0.0f);
+        thisx->textId = 0x703D;
+        this->actionFunc = func_80B4AF18;
+    } else if (Flags_GetEventChkInf(0x40)) {
+        frameCount = SkelAnime_GetFrameCount(&D_06000438.genericHeader);
+        SkelAnime_ChangeAnim(&this->skelAnime, &D_06000438, 1.0f, 0.0f, frameCount, 0, 0.0f);
+        thisx->textId = 0x703C;
+        this->actionFunc = func_80B4AF18;
+    } else {
+        thisx->textId = 0xFFFF;
+        this->actionFunc = func_80B4B010;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl1/func_80B4AF18.s")
+void EnZl1_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnZl1* this = THIS;
 
+    SkelAnime_Free(&this->skelAnime, globalCtx);
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void func_80B4AE18(EnZl1* this) {
+    if ((this->skelAnime.animCurrentSeg == &D_06010B38) && (this->skelAnime.animCurrentFrame < 26.0f)) {
+        this->unk_1F4 = &D_06008C48;
+        this->unk_1F8 = &D_06008848;
+        this->unk_1FC = 2;
+        return;
+    }
+
+    if (DECR(this->unk_1FC) == 0) {
+        this->unk_1FC = Math_Rand_S16Offset(0x1E, 0xA);
+    }
+
+    if (this->unk_1FC < 4) {
+        this->unk_1FE = this->unk_1FC;
+    } else {
+        this->unk_1FE = 0;
+    }
+
+    this->unk_1F4 = D_80B4E61C[this->unk_1FE];
+    this->unk_1F8 = D_80B4E61C[this->unk_1FE];
+    this->unk_1EC = D_80B4E62C[this->unk_1F2];
+}
+
+void func_80B4AF18(Actor* thisx, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    EnZl1* this = THIS;
+
+    func_80038290(globalCtx, thisx, &this->unk_200, &this->unk_206, thisx->posRot2.pos);
+
+    if (this->unk_1E6 != 0) {
+        if (func_8002F334(thisx, globalCtx) != 0) {
+            this->unk_1E6 = 0;
+        }
+    } else {
+        if (func_8002F194(thisx, globalCtx)) {
+            this->unk_1E6 = 1;
+        } else {
+            if (thisx->posRot.pos.y <= player->actor.posRot.pos.y) {
+                func_8002F2F4(thisx, globalCtx);
+            }
+        }
+    }
+
+    Collider_CylinderUpdate(&this->actor, &this->collider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+}
+
+#ifdef NON_MATCHING
+// regalloc issues with the envCtx.unk_E2 section - using v0 instead of v1
+// silly if (rotDiff) line added because it fixes almost every regalloc issue
+void func_80B4B010(Actor* thisx, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
```

Comment:
why is this taking a base actor thisx? its not an entry point function. this might help with your non matching issues